### DISCUSSION TITLE
Max connections error

### DISCUSSION THREAD

**Discussion by inteckltd** [supabase/supabase#22430] [Questions]
I have recently in the past couple of days been receiving an error periodically whereby my system crashes and reports `max-connections`.

I have setup a Grafana dashboard and can see the active connections constantly increasing overtime to the point of the crash occurring.

![Screenshot 2024-04-03 at 11 25 46](https://github.com/supabase/supabase/assets/55838611/b7b0a3c9-6361-4013-8c48-2219770013ce)

As you can see there was a drop of connections at around 10:15 and this was when I manually restarted our GoLang API, which leads me to believe the connection between my API and Supabase is not setup correctly. As this has only just recently started happening, and there has been changes recently with Connection Pooling, I feel like these are connected in some way.

I am currently setup to use the new Connection Pooling on the dashboard which uses port `5432`.

![Screenshot 2024-04-03 at 11 29 10](https://github.com/supabase/supabase/assets/55838611/93cc133e-46c4-4a60-a2f0-854530f00473)

On my API I use a long running connection, so there is never a new connection created - There should only ever be 1 single connection that gets created on initialisation:

![Screenshot 2024-04-03 at 11 30 17](https://github.com/supabase/supabase/assets/55838611/e049b741-df6e-4993-80a6-8baf862d6c1a)

I did see this guide here: https://github.com/orgs/supabase/discussions/22305 and the results I feel are below:

1. There is only around 100 active users on this system so I don't feel like the compute size (which is currently small) is the issue.
2. Due to the users on the system, even though the query time could be decreased, I don't feel this is sufficient enough to be causing the connections.
3. Pool size has been manually increased to 60 from 15 on my small compute instance, again, as the connections are constantly growing this will eventually get hit.

**Comment by Xchanger25** [comment]
Built a small thing for this if it's useful.. watches four counters and emails
once when one spikes: https://github.com/Xchanger25/watchbill
